firebase-inappmessaging/src/main/java/com/google/firebase/inappmessaging/internal/InAppMessageStreamManager.java

```diff
@@ -197,7 +197,14 @@ private boolean shouldIgnoreCache(String event) {
                                       Logging.logi(
                                           String.format(
                                               "Already impressed %s ? : %s",
```

Comment:
Lets do `Already impression campaign / experiment` appropriately... Also, lets move this log line generation to a function.